Sming/Arch/Esp8266/Components/heap/component.mk

```diff
@@ -9,13 +10,13 @@ COMPONENT_SUBMODULES	:= umm_malloc
 COMPONENT_SRCFILES		+= custom_heap.c umm_malloc/src/umm_malloc.c
 COMPONENT_INCDIRS		+= umm_malloc/src umm_malloc/includes/c-helper-macros
 
-COMPONENT_DEPENDS		:= esp8266
-
 COMPONENT_VARS			+= UMM_POISON_CHECK
 ifeq ($(UMM_POISON_CHECK),1)
 GLOBAL_CFLAGS			+= -DUMM_POISON_CHECK
 endif
 
+COMPONENT_CFLAGS		+= -Wno-array-bounds
```

Comment:
Why is that change needed?

Reply:
If you comment this out then try compiling with `ENABLE_CUSTOM_HEAP=1`:

```
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c: In function 'umm_init':
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:73:32: error: array subscript 1 is outside array bounds of 'char[1]' [-Werror=array-bounds]
   73 | #define UMM_BLOCK(b)  (umm_heap[b])
      |                       ~~~~~~~~~^~~~
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:75:24: note: in expansion of macro 'UMM_BLOCK'
   75 | #define UMM_NBLOCK(b) (UMM_BLOCK(b).header.used.next)
      |                        ^~~~~~~~~
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:231:5: note: in expansion of macro 'UMM_NBLOCK'
  231 |     UMM_NBLOCK(block_1th) = block_last | UMM_FREELIST_MASK;
      |     ^~~~~~~~~~
In file included from s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:35:
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc_cfg.h:60:13: note: while referencing '_heap_start'
   60 | extern char _heap_start;
      |             ^~~~~~~~~~~
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:73:32: error: array subscript 1 is outside array bounds of 'char[1]' [-Werror=array-bounds]
   73 | #define UMM_BLOCK(b)  (umm_heap[b])
      |                       ~~~~~~~~~^~~~
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:77:24: note: in expansion of macro 'UMM_BLOCK'
   77 | #define UMM_NFREE(b)  (UMM_BLOCK(b).body.free.next)
      |                        ^~~~~~~~~
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:232:5: note: in expansion of macro 'UMM_NFREE'
  232 |     UMM_NFREE(block_1th)  = 0;
      |     ^~~~~~~~~
In file included from s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:35:
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc_cfg.h:60:13: note: while referencing '_heap_start'
   60 | extern char _heap_start;
      |             ^~~~~~~~~~~
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:73:32: error: array subscript 1 is outside array bounds of 'char[1]' [-Werror=array-bounds]
   73 | #define UMM_BLOCK(b)  (umm_heap[b])
      |                       ~~~~~~~~~^~~~
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:76:24: note: in expansion of macro 'UMM_BLOCK'
   76 | #define UMM_PBLOCK(b) (UMM_BLOCK(b).header.used.prev)
      |                        ^~~~~~~~~
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:233:5: note: in expansion of macro 'UMM_PBLOCK'
  233 |     UMM_PBLOCK(block_1th) = block_0th;
      |     ^~~~~~~~~~
In file included from s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:35:
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc_cfg.h:60:13: note: while referencing '_heap_start'
   60 | extern char _heap_start;
      |             ^~~~~~~~~~~
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:73:32: error: array subscript 1 is outside array bounds of 'char[1]' [-Werror=array-bounds]
   73 | #define UMM_BLOCK(b)  (umm_heap[b])
      |                       ~~~~~~~~~^~~~
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:78:24: note: in expansion of macro 'UMM_BLOCK'
   78 | #define UMM_PFREE(b)  (UMM_BLOCK(b).body.free.prev)
      |                        ^~~~~~~~~
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:234:5: note: in expansion of macro 'UMM_PFREE'
  234 |     UMM_PFREE(block_1th)  = block_0th;
      |     ^~~~~~~~~
In file included from s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc.c:35:
s:/sandboxes/sming-dev/Sming/Arch/Esp8266/Components/heap/umm_malloc/src/umm_malloc_cfg.h:60:13: note: while referencing '_heap_start'
   60 | extern char _heap_start;
      |             ^~~~~~~~~~~
cc1.exe: all warnings being treated as errors
```